src/azure-cli-core/azure/cli/core/_help.py

```diff
@@ -80,13 +79,75 @@ def _print_extensions_msg(help_file):
             if help_file.command_source.preview:
                 logger.warning(help_file.command_source.get_preview_warn_msg())
 
-    def print_detailed_help(self, cli_name, help_file):
+    def _print_detailed_help(self, cli_name, help_file):
         AzCliHelp._print_extensions_msg(help_file)
-        self._print_detailed_help(cli_name, help_file)
+        super(AzCliHelp, self)._print_detailed_help(cli_name, help_file)
+
+    @staticmethod
+    def update_parser_with_help_file(nouns, cmd_loader_map, parser, is_group):
+        import inspect
+        import os
+
+        def _parse_yaml_from_string(text, help_file_path):
+            import yaml
+
+            dir_name, base_name = os.path.split(help_file_path)
+
+            pretty_file_path = os.path.join(os.path.basename(dir_name), base_name)
+
+            try:
+                data = yaml.load(text)
+                if not data:
+                    raise CLIError("Error: Help file {} is empty".format(pretty_file_path))
+                return data
+            except yaml.YAMLError as e:
+                raise CLIError("Error parsing {}:\n\n{}".format(pretty_file_path, e))
+
+        command_nouns = " ".join(nouns)
+        loader = None
+        if is_group:
+            for k, v in cmd_loader_map.items():
+                if k.startswith(command_nouns):
+                    loader = v[0]
+                    break
+        else:
+            loader = cmd_loader_map.get(command_nouns, [None])[0]
+
+        if loader:
+            loader_file_path = inspect.getfile(loader.__class__)
+            dir_name = os.path.dirname(loader_file_path)
+            files = os.listdir(dir_name)
+            for file in files:
+                if file.endswith(".yaml") or file.endswith(".yml"):
```

Comment:
```python
if file.endswith((".yaml", ".yml")):
```

> Return True if the string ends with the specified suffix, otherwise return False. ***suffix can also be a tuple of suffixes to look for***. With optional start, test beginning at that position. With optional end, stop comparing at that position.

* https://docs.python.org/3/library/stdtypes.html#str.endswith